### ISSUE TITLE
Koopman() now supports infinite integration domains

### ISSUE THREAD

**Issue by djinnome** [SciML/SciMLTutorials.jl#530]
According to the SciML tutorial, 

https://github.com/SciML/SciMLTutorials.jl/blob/e8c8c3e51703fd08b874abbbc7b52ed895d0cb6f/tutorials/DiffEqUncertainty/01-expectation_introduction.jmd#L111

However, when I actually tried this, I got the right answer:

```julia
u0_dist = [Normal(3.0,2.0)]
expectation(g, prob, u0_dist, p, Koopman(), Tsit5())
```

```julia
u: 1-element Vector{Float64}:
 0.9035426476099575
```

Compared with the analytical solution:

```julia
exp(p[1]*4.0)*mean(u0_dist[1])
```

```julia
0.9035826357366064
```

**Comment by djinnome** [comment]
I am running Julia 1.8 with Plots v1.31.7, DifferentialEquations v7.2.0 and SciMLBase v1.51.1

**Comment by ChrisRackauckas** [comment]
yeah it's getting a lot of work. Good to hear this is now fixed. Moving this to SciMLExpectations where it will be doctested as https://scimlexpectations.sciml.ai/dev/tutorials/introduction/
